src/kv/encryptor.h

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
This contains most of what was `node/encryptor.h`. There are 2 notable changes:
- it knows nothing about `LedgerSecrets` types. It is instead constructed from a `KeyInfo` list, and calls a virtual `record_compacted_keys` for the reverse communication
- the different ways of setting iv_id (`set_view` in PBFT, node ID at construction in Raft) are combined into a single `set_iv_id` method on this base class. So Raft should always to remember call `set_iv_id` afterwards, once